From a4aa2ddf143879f0f9a4c89f130437ac960ca831 Mon Sep 17 00:00:00 2001
From: "Patil, Rachna" <rachna@ti.com>
Date: Mon, 23 Jan 2012 16:56:59 +0530
Subject: [PATCH 0756/1092] arm: omap: am33xx: update TSC hwmod data

Sysconfig register bits are updated for TSC
hwmod. TSC supports idle mode at bit position 2&3,
hence following omap4 IP's.

Signed-off-by: Patil, Rachna <rachna@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index f9b91fa..77030e4 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -277,8 +277,18 @@ static struct omap_hwmod am33xx_l4wkup_hwmod = {
 };
 
 /* 'adc_tsc' class */
+static struct omap_hwmod_class_sysconfig am33xx_adc_tsc_sysc = {
+	.rev_offs	= 0x00,
+	.sysc_offs	= 0x10,
+	.sysc_flags	= SYSC_HAS_SIDLEMODE,
+	.idlemodes	= (SIDLE_FORCE | SIDLE_NO |
+				SIDLE_SMART | SIDLE_SMART_WKUP),
+	.sysc_fields	= &omap_hwmod_sysc_type2,
+};
+
 static struct omap_hwmod_class am33xx_adc_tsc_hwmod_class = {
 	.name		= "adc_tsc",
+	.sysc		= &am33xx_adc_tsc_sysc,
 };
 
 /* adc_tsc */
-- 
1.7.11.2

